### PR TITLE
[qsharp/en-us] Update qsharp example

### PR DESCRIPTION
**Pull request by sezna**
- [x] I solemnly swear that this is all original content of which I am the original author
- [x] Pull request title is prepended with `[language/lang-code]` (example `[python/fr-fr]` or `[java/en]`)
- [x] Pull request touches only one file (or a set of logically related files with similar changes made)
- [x] Content changes are aimed at *intermediate to experienced programmers* (this is a poor format for explaining fundamental programming concepts)
- [N/A] If you've changed any part of the YAML Frontmatter, make sure it is formatted according to [CONTRIBUTING.md](https://github.com/adambard/learnxinyminutes-docs/blob/master/CONTRIBUTING.markdown)
  - [x] Yes, I have double-checked quotes and field names!

The Q# language has evolved a bit since this example was last updated. This PR updates the example code to compile in the modern compiler, runs the canonical formatter on it, updates the syntax highlighting to the now-supported `qsharp` syntax highlighter, and updates the references to the latest content we've published.

### CODE REVIEW

File: qsharp.html.markdown [hunk 1]
```diff
@@ -4,47 +4,49 @@ contributors:
...
     - ["Mariia Mykhailova", "https://github.com/tcNickolas"]
     - ["Andrew Ryan Davis", "https://github.com/AndrewDavis1191"]
+    - ["Alex Hansen", "https://github.com/sezna"]
 filename: LearnQSharp.qs
 ---
 
 Q# is a high-level domain-specific language which enables developers to write quantum algorithms. Q# programs can be executed on a quantum simulator running on a classical computer and (in future) on quantum computers.
 
-```c#
+```qsharp
```
Comment (verhovsky) [inline qsharp.html.markdown:13]: `qsharp` is not supported by Pygments (the syntax highlighter used on Learn X in Y Minutes)

https://pygments.org/languages/

so either disable syntax highlighting by not including any language name here, or stick with C# or whatever is the closest matching language

File: qsharp.html.markdown [hunk 2]
```diff
@@ -58,97 +60,102 @@ using (qs = Qubit[2]) {
 /////////////////////////////////////
 // 2. Classical data types and operators
 
-// Numbers in Q# can be stored in Int, BigInt or Double.
-let i = 1;            // This defines an Int variable i equal to 1
-let bi = 1L;          // This defines a BigInt variable bi equal to 1
-let d = 1.0;          // This defines a Double variable d equal to 1
+function ClassicalDataTypes() : Unit {
```
Comment (verhovsky) [inline qsharp.html.markdown:63]: Is there a way to not put everything in functions? It's annoying to permanently take up screen space for an indent, especially if you're reading the article on a phone that would be like 20% of your entire screen

Comment (sezna) [reply qsharp.html.markdown:63]: I was trying to make the whole file runnable. As are no global-scope statements, that meant wrapping things. Is it ok for learn x in y if the code snippet doesn't actually work? If so, we could just remove the function declarations.

Comment (sezna) [reply qsharp.html.markdown:63]: Hi @verhovsky  -- I did some digging and it does look like other samples try to preserve the "runnability" of the code snippet ([java example](https://github.com/adambard/learnxinyminutes-docs/blob/master/java.html.markdown)). Just wondering if you have any more thoughts on this.

Comment (verhovsky) [reply qsharp.html.markdown:63]: nope, sounds good to me

**Comment by verhovsky** [comment]
Thank you for working on this

**Comment by sezna** [comment]
no problem, it was reasonably self-serving 😄  thanks for providing this educational resource.

**Pull request merged**